### ISSUE TITLE
Feature request: Add option to quickly unblock (noop) the domain with the most blocked network requests

### ISSUE THREAD

**Issue by WalterWW** [uBlock-LLC/uBlock#871]
When visiting a new site with 3rd-party requests blocked by default, it often takes a bit of trial and error to figure out which 3rd-party sites are needed to get the site to display properly. 

Requestpolicy addressed this by sorting the list of blocked domains by number of blocked requests. So simply allowing the first domain in the list fixed most sites.

It would be nice if ublock had a feature that allows adding a temporay noop rule for the domain with the highest number of blocked network requests by simply clicking a single button. Maybe it could even iterate through all the domains, so clicking the button more than once just adds the next highest domain etc., until the site displays properly.

This feature would be especially useful when the user is not interested in setting up permanent rules for a certain site, because he's only visiting it once anyway. Right now the only options are to go through all the trial and error or temporarily allow all requests from that site. Both are undesirable.

**Comment by WalterWW** [comment]
So I thought about this some more, and maybe my initial idea of a single button to unblock based on blocked requests isn't so ideal from a user perspective after all: In order to know which domain the button unblocks, the user would have to click it first. Not very transparent!

But how about this: ublock could add a visual indicator on the left side next to the domain with the highest number of blocked network requests, such as a little icon. And when the user clicks on that icon, it adds a temporary noop rule for that domain, and the icon jumps to the next highest domain, etc.

That way getting a site to work quickly would become very easy, as the user simply has to click on that icon until the site displays correctly. And at the same time it's always transparent for the user which domains are being unblocked.

**Comment by gorhill** [comment]
Let's take a step back here. I will be **extremely** conservative in keeping adding visual "noise" to the UI, where often less is better in order to not drown the important information in a sea of gadgets.

Here is what I think so far: it just works well now. Trying to have uBlock do everything for you is going to clutter the UI and turn it into a mess. So far un-breaking sites in default-deny has been rather easy I find, especially if you stick to root domain-level rules.

In short, I don't intend to add more at this point.

But I will take my time, and think about the suggestions, let the ideas mature.

**Comment by WalterWW** [comment]
Fair enough.

I guess I'll just have to change my habits then :)

**Comment by gorhill** [comment]
I added a line at the end in my comment.

**Comment by WalterWW** [comment]
You're probably right about the visual noise. It's easy to get used to new features if you're already familiar with all the old ones, but for new users too much could very quickly become overwhelming.

Mabye as a compromise you could consider adding an option for advanced users to choose between two different sorting modes for the domain list: Alphabetically or by number of blocked requests.

**Comment by mike-to-spec** [comment]
It sounds like this could be mitigated a bit from this issue #68: by allowing us to create rules from the network requests. You'd be able to see the most common request from there.

**Comment by WalterWW** [comment]
@MikhailTNY: Kinda, but not exactly. As a long time user of Requestpolicy I formed a habit of quickly clicking "temporarily allow" for the topmost domain in the blocked list every time I visited a new site that didn't display correctly, and where I had no intentions of setting up permanent rules. This fixed the site in maybe 90% of the time, because the domain list was sorted by number of blocked requests. 

I tried to bring this habit over to ublock and suggested a feature to improve on it with some automated helper. But I think gorhill is right about not adding too many unneccessary bells and whistles for features that are not really essential. I would actually prefer an option to simply sort the list by number of blocked requests now.
